lib/ex_format/state.ex

```diff
@@ -0,0 +1,29 @@
+defmodule ExFormat.State do
```

Comment:
Make this a struct instead of a map so that we have compile-time guarantees on its keys.

```elixir
defstruct [
  parenless_calls: MapSet.new(@parenless_calls),
  parenless_zero_arity?: false,
  in_spec: nil,
  ...,
]
```

Then you can remove `initialize_state/0` and just use `%ExFormat.State{}` when you need to initialize the state.